homeassistant/components/boiler/__init__.py

```diff
@@ -0,0 +1,496 @@
+})
+SET_OPERATION_MODE_SCHEMA = vol.Schema({
+    vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
+    vol.Required(ATTR_OPERATION_MODE): cv.string,
+})
+
+
+def set_away_mode(hass, away_mode, entity_id=None):
+    """Turn all or specified boiler controller devices away mode on."""
+    data = {
+        ATTR_AWAY_MODE: away_mode
+    }
+    # Add duration to work out new idle temp
+    # e.g. off to work, out to dinner or overnight
+
+    if entity_id:
+        data[ATTR_ENTITY_ID] = entity_id
+
+    hass.services.call(DOMAIN, SERVICE_SET_AWAY_MODE, data)
+
+
+def set_guest_mode(hass, guest_mode, total_guests, entity_id=None):
+    """Turn all or specified boiler controller devices guest mode on."""
+    data = {
+        ATTR_GUEST_MODE: guest_mode,
+        ATTR_TOTAL_GUESTS: total_guests
+    }
+
+    if entity_id:
+        data[ATTR_ENTITY_ID] = entity_id
+
+    hass.services.call(DOMAIN, SERVICE_SET_GUEST_MODE, data)
+
+
+def set_holiday_mode(hass, holiday_mode, holiday_duration, entity_id=None):
+    """Turn all or specified boiler controller devices holiday mode on."""
+    # Can away mode be used for this?
+    data = {
+        ATTR_HOLIDAY_MODE: holiday_mode,
+        ATTR_HOLIDAY_DURATION: holiday_duration
+    }
+
+    if entity_id:
+        data[ATTR_ENTITY_ID] = entity_id
+
+    hass.services.call(DOMAIN, SERVICE_SET_HOLIDAY_MODE, data)
+
+
+# pylint: disable=too-many-arguments
+def set_temperature(hass, entity_id=None,
+                    target_water_temperature=None,
+                    panel_diff_temp=None):
+    """
+    Set new target temperatures.
+
+    Used when either new target temp or panel differential temp is required.
+    """
+    kwargs = {
+        key: value for key, value in [
+            (ATTR_TARGET_WATER_TEMPERATURE, target_water_temperature),
+            (ATTR_PANEL_DIFF_TEMP, panel_diff_temp),
+            (ATTR_ENTITY_ID, entity_id),
+        ] if value is not None
+    }
+    _LOGGER.debug("set_temperature start data=%s", kwargs)
+    hass.services.call(DOMAIN, SERVICE_SET_WATER_TEMP, kwargs)
+
+
+def set_operation_mode(hass, operation_mode, entity_id=None):
+    """Set new target operation mode."""
+    data = {ATTR_OPERATION_MODE: operation_mode}
+
+    if entity_id is not None:
+        data[ATTR_ENTITY_ID] = entity_id
+
+    hass.services.call(DOMAIN, SERVICE_SET_OPERATION_MODE, data)
+
+
+# pylint: disable=too-many-branches
+def setup(hass, config):
+    """Setup boiler/geyser controller."""
+    component = EntityComponent(_LOGGER, DOMAIN, hass, SCAN_INTERVAL)
+    component.setup(config)
+
+    descriptions = load_yaml_config_file(
+        os.path.join(os.path.dirname(__file__), 'services.yaml'))
+
+    def away_mode_set_service(service):
+        """Set away mode on target boiler controller devices."""
+        target_boiler = component.extract_from_service(service)
+
+        away_mode = service.data.get(ATTR_AWAY_MODE)
+
+        if away_mode is None:
```

Comment:
No need to check this since you have it marked required in the voluptuous schema for this service.